HKMP/Api/Eventing/EventBase.cs

```diff
@@ -0,0 +1,52 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Hkmp.Eventing;
+
+namespace Hkmp.Api.Eventing
+{
+    /// <summary>
+    /// Base type for all events
+    /// </summary>
+    public class EventBase
+    {
+        private readonly List<EventSubscription> _subscriptions = new List<EventSubscription>();
+
+        /// <summary>
+        /// Unsubscribes an action from the event by it's token
+        /// </summary>
+        /// <param name="token">the token to unsubscribe</param>
+        public virtual void Unsubscribe(SubscriptionToken token)

```

Comment:
The parameter here is unused. Surely you want to pick the `EventSubscription` from the list that matches the token?

Reply:
Whoops, forgot the callback on line 21. Fixing.

Reply:
Is it the Typename only being T? I mean I can fix those real quick but I gather you mean more than that if I'm not adhering in most places.

Reply:
Sorry for being vague about it, the style I was missing in most places was the brace ('{') style. In the rest of the project all braces are at the end of the line rather than on a new line. I'll make a commit where I go over the code a reformat it a bit to my liking!